src/drive/web/modules/actions/useActions.spec.jsx

```diff
@@ -0,0 +1,422 @@
+    const { result } = renderActionsHook(...hookArgs)
+    return result.current[actionKey]
+  }
+
+  it('returns actions keyed by icon', () => {
+    const { result } = renderActionsHook(mockFolderId, {
+      canMove: true
+    })
+    expect(Object.keys(result.current)).toEqual([
+      'share',
+      'download',
+      'trash',
+      'openWith',
+      'rename',
+      'moveto',
+      'qualify',
+      'history',
+      'phone-download'
+    ])
+  })
+
+  describe('share action', () => {
+    it('is only visible with write access and a single selected item', () => {
+      mockSharingContextValue.hasWriteAccess.mockReturnValue(false)
+      const shareActionWithoutWriteAccess = getAction('share')
+      expect(shareActionWithoutWriteAccess.displayCondition(['abc'])).toBe(
+        false
+      )
+      expect(
+        shareActionWithoutWriteAccess.displayCondition(['abc', 'def'])
+      ).toBe(false)
+
+      mockSharingContextValue.hasWriteAccess.mockReturnValue(true)
+      const shareAction = getAction('share')
+      expect(shareAction.displayCondition(['abc'])).toBe(true)
+      expect(shareAction.displayCondition(['abc', 'def'])).toBe(false)
+    })
+
+    it('shows the share modal when activated', () => {
+      const shareAction = getAction('share')
+      const mockDocument = { id: 'abc', name: 'my-file.md' }
+      shareAction.action([mockDocument])
+      expect(mockModalContextValue.pushModal).toHaveBeenCalledWith(
+        <ShareModal
+          document={mockDocument}
+          documentType="Files"
+          sharingDesc="my-file.md"
+        />
+      )
+    })
+  })
+
+  describe('download action', () => {
+    describe('desktop', () => {
+      it('triggers a file download', () => {
+        const downloadAction = getAction('download')
+        const mockDocuments = [
+          { id: 'abc', name: 'my-file.md' },
+          { id: 'def', name: 'my-file-2.md' }
+        ]
+
+        downloadAction.action(mockDocuments)
+        expect(downloadFiles).toHaveBeenCalledWith(mockDocuments, mockClient)
+      })
+    })
+
+    describe('mobile', () => {
+      beforeEach(() => {
+        global.__TARGET__ = 'mobile'
+      })
+
+      it('is visible for a single file on iOS', () => {
+        global.window.cordova = { platformId: 'ios' }
+        const downloadAction = getAction('download')
+        expect(
+          downloadAction.displayCondition([{ id: 'abc', type: 'file' }])
+        ).toBe(true)
+        expect(
+          downloadAction.displayCondition([{ id: 'abc', type: 'folder' }])
+        ).toBe(false)
+        expect(
+          downloadAction.displayCondition([
+            { id: 'abc', type: 'file' },
+            { id: 'def', type: 'file' }
+          ])
+        ).toBe(false)
+        expect(
+          downloadAction.displayCondition([
+            { id: 'abc', type: 'file' },
+            { id: 'def', type: 'folder' }
+          ])
+        ).toBe(false)
+        global.window.cordova = null
```

Comment:
global.window.cordova will not be cleaned up if the test fail, prefer to setup/cleanup in beforeEach, afterEach.